### ISSUE TITLE
Why does Pyway not take the config params from config file provided by -c/--config arguments?

### ISSUE THREAD

**Issue by ChrisDAT20** [jasondcamp/pyway#73]
I wrote a file with connection parameters. No ENV variables set. There the database_port is given as 5433, which is a mapped docker port to a Postgres DB.

```
database_type: postgres
database_host: localhost
database_port: 5433
database_name: postgres
database_username: postgres
database_password: password
database_migration_dir: resources
database_table: public.pyway
```

However, when running the command, Pyway tries to access the DB via port 5432.
```
(venv) C:\Users\user1\PycharmProjects\project\db>pyway --config pyway.timeseries.conf info
PyWay 0.3.26
Gathering info...
Traceback (most recent call last):
  File "C:\Users\user1\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\user1\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\user1\PycharmProjects\project\venv\Scripts\pyway.exe\__main__.py", line 7, in <module>
  File "C:\Users\user1\PycharmProjects\project\venv\lib\site-packages\pyway\scripts\main.py", line 69, in cli
    info(config)
  File "C:\Users\user1\PycharmProjects\project\venv\lib\site-packages\pyway\scripts\main.py", line 33, in info
    tbl = Info(config).run()
  File "C:\Users\user1\PycharmProjects\project\venv\lib\site-packages\pyway\info.py", line 14, in __init__
    self._db = factory(config.database_type)(config)
  File "C:\Users\user1\PycharmProjects\project\venv\lib\site-packages\pyway\dbms\postgres.py", line 28, in __init__
    self.create_version_table_if_not_exists()
  File "C:\Users\user1\PycharmProjects\project\venv\lib\site-packages\pyway\dbms\postgres.py", line 34, in create_version_table_if_not_exists
    self.execute(CREATE_VERSION_MIGRATIONS % self.version_table)
  File "C:\Users\user1\PycharmProjects\project\venv\lib\site-packages\pyway\dbms\postgres.py", line 37, in execute
    conn = self.connect()
  File "C:\Users\user1\PycharmProjects\project\venv\lib\site-packages\pyway\dbms\postgres.py", line 31, in connect
    return psycopg2.connect(f"dbname={self.args.database_name} user={self.args.database_username} host={self.args.database_host} password={self.args.database_password} port={self.args.database_port}")  # noqa: E501
  File "C:\Users\user1\PycharmProjects\project\venv\lib\site-packages\psycopg2\__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: connection to server at "localhost" (::1), port 5432 failed: FATAL:  password authentication failed for user "postgres"


(venv) C:\Users\user1\PycharmProjects\project\db>
```

**Comment by ChrisDAT20** [comment]
It seems that a `.pyway.conf` does not allow the usage of -c/--config arguments. Ugly :)

Even uglier:
```
(venv) C:\Users\user1\PycharmProjects\project\db>pyway --config pyway.timeseries.conf info                                                 
PyWay 0.3.26
Gathering info...
Traceback (most recent call last):
  File "C:\Users\user1\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\user1\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\user1\PycharmProjects\project\venv\Scripts\pyway.exe\__main__.py", line 7, in <module>
  File "C:\Users\user1\PycharmProjects\project\venv\lib\site-packages\pyway\scripts\main.py", line 69, in cli
    info(config)
  File "C:\Users\user1\PycharmProjects\project\venv\lib\site-packages\pyway\scripts\main.py", line 33, in info
    tbl = Info(config).run()
  File "C:\Users\user1\PycharmProjects\project\venv\lib\site-packages\pyway\info.py", line 14, in __init__
    self._db = factory(config.database_type)(config)
TypeError: 'NoneType' object is not callable

(venv) C:\Users\user1\PycharmProjects\project\db>
```

Whats this? Does it mean that Pyway now does not know that we are on Postgres?

**Comment by jasondcamp** [comment]
I see the issue, we want cmd line arguments and env vars to override the config file, so it processes the arguments after the config file, preventing use of the --config option. Will fix in the next release, thanks!

**Comment by jasondcamp** [comment]
This should be fixed in 0.3.30 - let me know if you still are having issues.

**Comment by ChrisDAT20** [comment]
Confirmed as fixed with 0.3.30. Thank you for the very fast response!
